scene/3d/collision_shape.cpp

```diff
@@ -83,7 +83,6 @@ void CollisionShape::_notification(int p_what) {
 				if (shape.is_valid()) {
 					parent->shape_owner_add_shape(owner_id, shape);
 				}
-				_update_in_shape_owner();
```

Comment:
I've identified this change to be the root cause to godotengine/godot#45694.

I think the assumption here was that the _update_in_shape_owner() can be removed from NOTIFICATION_PARENTED, because it also gets called in NOTIFICATION_ENTER_TREE. However in case of godotengine/godot#45694 the function "Mesh Library -> Import from Scene..." instantiates the scene to be converted, but this scene never enters the scene tree. As a result, the collision shapes of the meshes in the mesh library don't have their proper transform.